### ISSUE TITLE
PHP7: Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP

### ISSUE THREAD

**Issue by vanmelick** [textpattern/textpattern#535]
Installing TXP using PHP 7 gives 10+ messages like this during step 4 of the setup:

> Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; IXR_Value has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 48

See https://wiki.php.net/rfc/remove_php4_constructors

Since we already require PHP4, I think we can safely change the constructor naming. Correct?

**Comment by vanmelick** [comment]
Might be useful. The complete list of errors I got. There may be more, bit it seems to be limited to the /lib directory

```
IXR_Value has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 48
IXR_Message has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 177
IXR_Server has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 375
IXR_Request has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 585
IXR_Client has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 629
IXR_Error has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 801
IXR_Date has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 843
IXR_Base64 has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 906
IXR_IntrospectionServer has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 927
IXR_ClientMulticall has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 1090
IXR_ClientSSL has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 1126
IXR_ClassServer has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/IXRClass.php on line 1350
PasswordHash has a deprecated constructor in /var/www/html/textpattern/textpattern/lib/PasswordHash.php on line 27
```

**Comment by vanmelick** [comment]
Do we have an upstream source for IXRClass.php?

**Comment by philwareham** [comment]
I've done a quick Google search and the [homepage is here](http://scripts.incutio.com/xmlrpc/) - doesn't look like this script has been updated since 2010 though (v1.7.4) and it's not available on GitHub either. Looks like it's fairly dead at this point in time.

Looking at GitHib options, there are a few options - [maybe this one?](https://github.com/milo/xml-rpc)

I'm not really sure of the value of this RPC stuff. Is it even needed?

**Comment by vanmelick** [comment]
Wordpress seems to have already fixed this:
https://github.com/WordPress/WordPress/commits/master/wp-includes/class-IXR.php

**Comment by philwareham** [comment]
I guess you could modify the IXRClass.php file as necessary to bring this up-to-date - it already contains Textpattern-specific code amendments anyway, and the original codebase is effectively dead.

**Comment by vanmelick** [comment]
Fixed by commit f1461b0
